lib/figaro.rb

```diff
@@ -0,0 +1,62 @@
+require 'yaml'
+require 'erb'
+
+class Figaro
+  def self.env
+    Environment
+  end
+
+  def self.setup(path)
+    Environment.setup(path)
+  end
+
+  def self.require_keys(keys)
+    Environment.require_keys(keys)
+  end
+
+  private
+end
+
+class Environment
+  def self.setup(path, env = Rails.env)
+    @config = {}
+    values = YAML.load(ERB.new(File.read(path)).result)
+
+    keys = Set.new(values.keys - ['production', 'development', 'test'])
+    keys |= values[env].keys
+
+    keys.each do |key|
+      value = values[env][key] || values[key]
+
+      if key == key.upcase
+        ENV[key] = value
+      end
```

Comment:
I think we need something like:

```
override_value = if ENV.key?(key)
  ENV[key]
elsif ENV.key?(key.upcase)
  ENV[key.upcase]
end
```

or some similar behavior that lets us override `application.yml` values via ENV. It's used in our Docker configs to override a few values:

https://github.com/18F/identity-idp/blob/master/docker-compose.yml#L21-L22

Reply:
Good catch, thank you. It seems counterintuitive to override the file-based config with system environment values

In checking the dev sandbox environment, it doesn't seem like those docker environment values are getting set, does this only affect local development? If so, is it possible to use the YAML config instead as the one source of truth?

Reply:
Looked into Figaro's behavior further, and it does prioritize loading from ENV over application.yml (and emits a warning). https://github.com/18F/identity-idp/pull/4408/commits/525b49ce946cffb8c17e9ee15177ea988e74d955 matches that behavior.